xchange-binance/src/main/java/info/bitrich/xchangestream/binance/BinanceStreamingMarketDataService.java

```diff
@@ -119,9 +120,21 @@ public BinanceStreamingMarketDataService(BinanceStreamingService service, Binanc
             ));
     }
 
-    private static String channelFromCurrency(CurrencyPair currencyPair, String subscriptionType) {
+    private String channelFromCurrency(CurrencyPair currencyPair, String subscriptionType) {
         String currency = String.join("", currencyPair.toString().split("/")).toLowerCase();
-        return currency + "@" + subscriptionType;
+        String currencyChannel = currency + "@" + subscriptionType;
+
+        if ("depth".equals(subscriptionType)) {
```

Comment:
It feels like this logic is now complex enough to warrant sharing between here and `BinanceStreamingExchange.buildSubscriptionStrings`.